beamer/events.py

```diff
@@ -214,8 +213,13 @@ def fetch(self) -> list[Event]:
                 result.extend(events)
                 from_block = BlockNumber(to_block + 1)
 
-        self._next_block_number = from_block
+        try:
+            # Block number needs to be decremented here, because it is already incremented above
+            block_data = self._contract.web3.eth.get_block(from_block - 1)
```

Comment:
We can instead use just `block_number`, which stays constant and is the loop end condition.

Reply:
No, because we might `break` in the while loop, this would be incorrect.